src/gui/EditWidgetIcons.cpp

```diff
@@ -223,8 +223,15 @@ void EditWidgetIcons::fetchFavicon(const QUrl& url)
     }
 
     m_httpClient->setConnectingTimeOut(5000, [this]() {
-        resetFaviconDownload();
-        MessageBox::warning(this, tr("Error"), tr("Unable to fetch favicon."));
+        if (m_url.startsWith("http://")) {
+            resetFaviconDownload();
+            MessageBox::warning(this, tr("Error"), tr("Unable to fetch favicon."));
+        } else {
+            m_url = m_url.replace("https","http");
```

Comment:
I was actually trying to fix this as well and I used QRegExp on the URL by matching this: `(https?://.+)$` it would pick the last URL, even match `cmd://firefox https://https.com/https/https.https` properly.